sources/packages/backend/apps/api/src/route-controllers/overaward/overaward.aest.controller.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  Controller,
+  Param,
+  ParseIntPipe,
+  Get,
+  Post,
+  Body,
+  NotFoundException,
+} from "@nestjs/common";
+import { ApiTags } from "@nestjs/swagger";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import {
+  AllowAuthorizedParty,
+  Groups,
+  Roles,
+  UserToken,
+} from "../../auth/decorators";
+import { UserGroups } from "../../auth/user-groups.enum";
+import { ClientTypeBaseRoute } from "../../types";
+import BaseController from "../BaseController";
+import { DisbursementOverawardService } from "@sims/services";
+import { StudentService } from "../../services";
+import {
+  OverawardBalanceAPIOutDTO,
+  OverawardAPIOutDTO,
+  OverawardManualRecordAPIInDTO,
+} from "./models/overaward.dto";
+import { PrimaryIdentifierAPIOutDTO } from "../models/primary.identifier.dto";
+import { IUserToken, Role } from "../../auth";
+
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("overaward")
+@ApiTags(`${ClientTypeBaseRoute.AEST}-overaward`)
+export class OverawardAESTController extends BaseController {
+  constructor(
+    private readonly disbursementOverawardService: DisbursementOverawardService,
+    private readonly studentService: StudentService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get the overaward balance of a student.
+   * @param studentId student.
+   * @returns overaward balance for student.
+   */
+  @Get("student/:studentId/balance")
+  async getOverawardBalance(
+    @Param("studentId", ParseIntPipe) studentId: number,
+  ): Promise<OverawardBalanceAPIOutDTO> {
+    const overawardBalance =
+      await this.disbursementOverawardService.getOverawardBalance([studentId]);
+
+    return { overawardBalanceValues: overawardBalance[studentId] };
```

Comment:
We may be crossing a line here where we are allowing the services object to be directly returned. If the service class receives more properties it will be directly exposed to the API. Maybe not be a big issue for the AEST but a possible issue when the students will be also receiving the data.

Reply:
I get your point here but because of overaward balance being dynamic `Record<string,number>` , how could we control retaining the dynamic behaviour in place? may be for students we need to used typed award codes in DTO?

Reply:
As we talked this is a key/value pair that does not seem likely to be expanded. I would not have a major concern to keep it as is for AEST and Ministry.